### ISSUE TITLE
[JENKINS-68122] Agent connection broken (randomly) with error java.util.concurrent.TimeoutException (regression in 2.325)

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-14#947]
<p>After upgrade Jenkins from 2.319.2 to 2.332.1 we start experience with EC2 agent connection broken with time out ping thread error:  </p>

<p>java.util.concurrent.TimeoutException: Ping started at 1648107727099 hasn't completed by 1648107967100java.util.concurrent.TimeoutException: Ping started at 1648107727099 hasn't completed by 1648107967100 at hudson.remoting.PingThread.ping(PingThread.java:132) at hudson.remoting.PingThread.run(PingThread.java:88)</p>

<p>This happen randomly and the build job is hung at pipeline git check out stage. When the agent connection broken, we can re-launch and it reconnect but the build job seem no longer can access the agent and just stall until cancel. While this happen, other EC2 agent still running and ping at os level from master to agent in question still get response. We tried to disable "Response Time" in Preventive Node monitoring (manage Node and Cloud). This just delay the broken connection from 2 missing ping to 5 or 6 as the master continue to monitor disk space, swap... Kill the job and rebuild will success most of the time (some time stuck on the same broken connection). </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kapawo">kapawo</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-68122" target="_blank">Agent connection broken (randomly) with error java.util.concurrent.TimeoutException (regression in 2.325)</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=basil">basil</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2022-04-09T15:22:22+00:00
<li><b>votes</b>: 5
<li><b>watchers</b>: 11
<li><b>imported</b>: 2025-11-14
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>After upgrade Jenkins from 2.319.2 to 2.332.1 we start experience with EC2 agent connection broken with time out ping thread error:  </p>

<p>java.util.concurrent.TimeoutException: Ping started at 1648107727099 hasn't completed by 1648107967100java.util.concurrent.TimeoutException: Ping started at 1648107727099 hasn't completed by 1648107967100 at hudson.remoting.PingThread.ping(PingThread.java:132) at hudson.remoting.PingThread.run(PingThread.java:88)</p>

<p>This happen randomly and the build job is hung at pipeline git check out stage. When the agent connection broken, we can re-launch and it reconnect but the build job seem no longer can access the agent and just stall until cancel. While this happen, other EC2 agent still running and ping at os level from master to agent in question still get response. We tried to disable "Response Time" in Preventive Node monitoring (manage Node and Cloud). This just delay the broken connection from 2 missing ping to 5 or 6 as the master continue to monitor disk space, swap... Kill the job and rebuild will success most of the time (some time stuck on the same broken connection). </p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins 2.332.1 on Ubuntu 18.04 with OpenJDk 11.0.14
Amazon EC2 plugin 1.68
```
</details>
<details><summary><i>4 attachments</i></summary>

- [java_deadlock_dump_1](https://issues.jenkins.io/secure/attachment/57625/java_deadlock_dump_1)
- [java_deadlock_dump_2](https://issues.jenkins.io/secure/attachment/57624/java_deadlock_dump_2)
- [java_deadlock_dump_3](https://issues.jenkins.io/secure/attachment/57623/java_deadlock_dump_3)
- [threaddump_showing_classloader_sync_hang.txt](https://issues.jenkins.io/secure/attachment/57640/threaddump_showing_classloader_sync_hang.txt)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER141913's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=423868&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-423868">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>jstack on failed agent found one dead lock</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
Found one Java-level deadlock:
=============================
<span class="code-quote">"RemoteInvocationHandler [#1]"</span>:
  waiting to lock monitor 0x00007fcf28002980 (object 0x0000000624c21510, a hudson.util.RingBufferLogHandler),
  which is held by <span class="code-quote">"Channel reader thread: channel"</span>
<span class="code-quote">"Channel reader thread: channel"</span>:
  waiting to lock monitor 0x00007fcf28006580 (object 0x0000000624c00ce0, a hudson.remoting.RemoteClassLoader),
  which is held by <span class="code-quote">"pool-1-thread-1 <span class="code-keyword">for</span> channel id=10591761"</span>
<span class="code-quote">"pool-1-thread-1 <span class="code-keyword">for</span> channel id=10591761"</span>:
  waiting to lock monitor 0x00007fcf28002980 (object 0x0000000624c21510, a hudson.util.RingBufferLogHandler),
  which is held by <span class="code-quote">"Channel reader thread: channel"</span>Java stack information <span class="code-keyword">for</span> the threads listed above:
===================================================
<span class="code-quote">"RemoteInvocationHandler [#1]"</span>:
        at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
        - waiting to lock &lt;0x0000000624c21510&gt; (a hudson.util.RingBufferLogHandler)
        at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:979)
        at java.util.logging.Logger.doLog(java.logging@11.0.14/Logger.java:1006)
        at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:1051)
        at hudson.remoting.RemoteInvocationHandler$Unexporter.reportStats(RemoteInvocationHandler.java:702)
        at hudson.remoting.RemoteInvocationHandler$Unexporter.run(RemoteInvocationHandler.java:594)
        at java.util.concurrent.Executors$RunnableAdapter.call(java.base@11.0.14/Executors.java:515)
        at java.util.concurrent.FutureTask.run(java.base@11.0.14/FutureTask.java:264)
        at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:121)
        at java.lang.<span class="code-object">Thread</span>.run(java.base@11.0.14/<span class="code-object">Thread</span>.java:829)
<span class="code-quote">"Channel reader thread: channel"</span>:
        at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
        - locked &lt;0x0000000624c21510&gt; (a hudson.util.RingBufferLogHandler)
        at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:979)
        at java.util.logging.Logger.doLog(java.logging@11.0.14/Logger.java:1006)
        at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:1092)
        at hudson.remoting.Channel$1.handle(Channel.java:608)
        at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:81)
<span class="code-quote">"pool-1-thread-1 <span class="code-keyword">for</span> channel id=10591761"</span>:
        at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
        - waiting to lock &lt;0x0000000624c21510&gt; (a hudson.util.RingBufferLogHandler)
        at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:979)
        at java.util.logging.Logger.doLog(java.logging@11.0.14/Logger.java:1006)
        at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:1092)
        at hudson.remoting.RemoteClassLoader.prefetchClassReference(RemoteClassLoader.java:387)
        - locked &lt;0x0000000624ad56f0&gt; (a java.util.Collections$SynchronizedMap)
        at hudson.remoting.RemoteClassLoader.loadWithMultiClassLoader(RemoteClassLoader.java:253)
        at hudson.remoting.RemoteClassLoader.findClass(RemoteClassLoader.java:223)
        at java.lang.<span class="code-object">ClassLoader</span>.loadClass(java.base@11.0.14/<span class="code-object">ClassLoader</span>.java:589)
        - locked &lt;0x0000000624c00ce0&gt; (a hudson.remoting.RemoteClassLoader)
        at java.lang.<span class="code-object">ClassLoader</span>.loadClass(java.base@11.0.14/<span class="code-object">ClassLoader</span>.java:522)
        at hudson.node_monitors.SwapSpaceMonitor$MonitorTask.call(SwapSpaceMonitor.java:123)
        at hudson.node_monitors.SwapSpaceMonitor$MonitorTask.call(SwapSpaceMonitor.java:118)
        at hudson.remoting.UserRequest.perform(UserRequest.java:211)
        at hudson.remoting.UserRequest.perform(UserRequest.java:54)
        at hudson.remoting.Request$2.run(Request.java:376)
        at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)
        at hudson.remoting.InterceptingExecutorService$$Lambda$44/0x0000000840098840.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(java.base@11.0.14/FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.14/ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.14/ThreadPoolExecutor.java:628)
        at java.lang.<span class="code-object">Thread</span>.run(java.base@11.0.14/<span class="code-object">Thread</span>.java:829)Found 1 deadlock.

</pre>
</div></div>
</pre>
</details>

</details>
<p>jstack on failed agent found one dead lock</p>

<pre>

Found one Java-level deadlock:
=============================
<span class="code-quote">"RemoteInvocationHandler [#1]"</span>:
  waiting to lock monitor 0x00007fcf28002980 (object 0x0000000624c21510, a hudson.util.RingBufferLogHandler),
  which is held by <span class="code-quote">"Channel reader thread: channel"</span>
<span class="code-quote">"Channel reader thread: channel"</span>:
  waiting to lock monitor 0x00007fcf28006580 (object 0x0000000624c00ce0, a hudson.remoting.RemoteClassLoader),
  which is held by <span class="code-quote">"pool-1-thread-1 <span class="code-keyword">for</span> channel id=10591761"</span>
<span class="code-quote">"pool-1-thread-1 <span class="code-keyword">for</span> channel id=10591761"</span>:
  waiting to lock monitor 0x00007fcf28002980 (object 0x0000000624c21510, a hudson.util.RingBufferLogHandler),
  which is held by <span class="code-quote">"Channel reader thread: channel"</span>Java stack information <span class="code-keyword">for</span> the threads listed above:
===================================================
<span class="code-quote">"RemoteInvocationHandler [#1]"</span>:
at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
- waiting to lock <0x0000000624c21510> (a hudson.util.RingBufferLogHandler)
at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:979)
at java.util.logging.Logger.doLog(java.logging@11.0.14/Logger.java:1006)
at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:1051)
at hudson.remoting.RemoteInvocationHandler$Unexporter.reportStats(RemoteInvocationHandler.java:702)
at hudson.remoting.RemoteInvocationHandler$Unexporter.run(RemoteInvocationHandler.java:594)
at java.util.concurrent.Executors$RunnableAdapter.call(java.base@11.0.14/Executors.java:515)
at java.util.concurrent.FutureTask.run(java.base@11.0.14/FutureTask.java:264)
at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:121)
at java.lang.<span class="code-object">Thread</span>.run(java.base@11.0.14/<span class="code-object">Thread</span>.java:829)
<span class="code-quote">"Channel reader thread: channel"</span>:
at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
- locked <0x0000000624c21510> (a hudson.util.RingBufferLogHandler)
at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:979)
at java.util.logging.Logger.doLog(java.logging@11.0.14/Logger.java:1006)
at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:1092)
at hudson.remoting.Channel$1.handle(Channel.java:608)
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:81)
<span class="code-quote">"pool-1-thread-1 <span class="code-keyword">for</span> channel id=10591761"</span>:
at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
- waiting to lock <0x0000000624c21510> (a hudson.util.RingBufferLogHandler)
at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:979)
at java.util.logging.Logger.doLog(java.logging@11.0.14/Logger.java:1006)
at java.util.logging.Logger.log(java.logging@11.0.14/Logger.java:1092)
at hudson.remoting.RemoteClassLoader.prefetchClassReference(RemoteClassLoader.java:387)
- locked <0x0000000624ad56f0> (a java.util.Collections$SynchronizedMap)
at hudson.remoting.RemoteClassLoader.loadWithMultiClassLoader(RemoteClassLoader.java:253)
at hudson.remoting.RemoteClassLoader.findClass(RemoteClassLoader.java:223)
at java.lang.<span class="code-object">ClassLoader</span>.loadClass(java.base@11.0.14/<span class="code-object">ClassLoader</span>.java:589)
- locked <0x0000000624c00ce0> (a hudson.remoting.RemoteClassLoader)
at java.lang.<span class="code-object">ClassLoader</span>.loadClass(java.base@11.0.14/<span class="code-object">ClassLoader</span>.java:522)
at hudson.node_monitors.SwapSpaceMonitor$MonitorTask.call(SwapSpaceMonitor.java:123)
at hudson.node_monitors.SwapSpaceMonitor$MonitorTask.call(SwapSpaceMonitor.java:118)
at hudson.remoting.UserRequest.perform(UserRequest.java:211)
at hudson.remoting.UserRequest.perform(UserRequest.java:54)
at hudson.remoting.Request$2.run(Request.java:376)
at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)
at hudson.remoting.InterceptingExecutorService$$Lambda$44/0x0000000840098840.call(Unknown Source)
at java.util.concurrent.FutureTask.run(java.base@11.0.14/FutureTask.java:264)
at java.util.concurrent.ThreadPoolExecutor.runWorker(java.base@11.0.14/ThreadPoolExecutor.java:1128)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(java.base@11.0.14/ThreadPoolExecutor.java:628)
at java.lang.<span class="code-object">Thread</span>.run(java.base@11.0.14/<span class="code-object">Thread</span>.java:829)Found 1 deadlock.

</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER141913's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424005&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424005">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Disable ping tread on both server and agent per <a href="https://wiki.jenkins.io/JENKINS/Ping-Thread.html" class="external-link" target="_blank" rel="nofollow noopener">document</a>, disable node monitor in Global Setting. Still got error randomly.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
This is a Unix agent
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by jenkins.slaves.StandardOutputSwapper$ChannelSwapper to constructor java.io.FileDescriptor(<span class="code-object">int</span>)
WARNING: Please consider reporting <span class="code-keyword">this</span> to the maintainers of jenkins.slaves.StandardOutputSwapper$ChannelSwapper
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a <span class="code-keyword">future</span> release
Evacuated stdout
Agent successfully connected and online
ERROR: Failed to monitor <span class="code-keyword">for</span> Architecture
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Swap Space
ERROR: Failed to monitor <span class="code-keyword">for</span> Clock Difference
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Response Time
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Disk Space
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Temp Space
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.ResponseTimeMonitor$1.monitor(ResponseTimeMonitor.java:56)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)ERROR: Failed to monitor <span class="code-keyword">for</span> Architecture
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Clock Difference
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Temp Space
ERROR: Failed to monitor <span class="code-keyword">for</span> Response Time
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: ERROR: Failed to monitor <span class="code-keyword">for</span> Free Disk Space
Failed to monitor <span class="code-keyword">for</span> Free Swap Space
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.ResponseTimeMonitor$1.monitor(ResponseTimeMonitor.java:56)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)

</pre>
</div></div>
</pre>
</details>

</details>
<p>Disable ping tread on both server and agent per <a href="https://wiki.jenkins.io/JENKINS/Ping-Thread.html" class="external-link" target="_blank" rel="nofollow noopener">document</a>, disable node monitor in Global Setting. Still got error randomly.</p>

<pre>

This is a Unix agent
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by jenkins.slaves.StandardOutputSwapper$ChannelSwapper to constructor java.io.FileDescriptor(<span class="code-object">int</span>)
WARNING: Please consider reporting <span class="code-keyword">this</span> to the maintainers of jenkins.slaves.StandardOutputSwapper$ChannelSwapper
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a <span class="code-keyword">future</span> release
Evacuated stdout
Agent successfully connected and online
ERROR: Failed to monitor <span class="code-keyword">for</span> Architecture
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Swap Space
ERROR: Failed to monitor <span class="code-keyword">for</span> Clock Difference
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Response Time
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Disk Space
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Temp Space
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.ResponseTimeMonitor$1.monitor(ResponseTimeMonitor.java:56)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)ERROR: Failed to monitor <span class="code-keyword">for</span> Architecture
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Clock Difference
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: Failed to monitor <span class="code-keyword">for</span> Free Temp Space
ERROR: Failed to monitor <span class="code-keyword">for</span> Response Time
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
ERROR: ERROR: Failed to monitor <span class="code-keyword">for</span> Free Disk Space
Failed to monitor <span class="code-keyword">for</span> Free Swap Space
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.ResponseTimeMonitor$1.monitor(ResponseTimeMonitor.java:56)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)
java.util.concurrent.TimeoutException
	at hudson.remoting.Request$1.get(Request.java:321)
	at hudson.remoting.Request$1.get(Request.java:240)
	at hudson.remoting.FutureAdapter.get(FutureAdapter.java:66)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitorDetailed(AbstractAsyncNodeMonitorDescriptor.java:112)
	at hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor.monitor(AbstractAsyncNodeMonitorDescriptor.java:76)
	at hudson.node_monitors.AbstractNodeMonitorDescriptor$Record.run(AbstractNodeMonitorDescriptor.java:305)

</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER142437's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424060&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424060">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>We're seeing what I believe is the same bug. It started after we upgraded to 2.332.1 last week.</p>

<p>19 out of 20 times the agent goes into a locked state sometime after it has connected to the master but before it receives a job to execute.</p>

<p>Pretty much a total showstopper for us, Jenkins is unusable at the moment.  Really hoping there is some workaround for this, like downgrading to the previous agent.jar perhaps?</p>

<p>I've taken a number of thread dumps of hanging agent.jar processes. They're all slightly different. What they all have in common is multiple threads stuck waiting:
 at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)</p>

<p>I'll upload a couple of the dumps, but they're really all much the same as what <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kapawo" class="user-hover" rel="kapawo">kapawo</a> has uploaded.</p>
</pre>
</details>

</details>
<p>We're seeing what I believe is the same bug. It started after we upgraded to 2.332.1 last week.</p>

<p>19 out of 20 times the agent goes into a locked state sometime after it has connected to the master but before it receives a job to execute.</p>

<p>Pretty much a total showstopper for us, Jenkins is unusable at the moment.  Really hoping there is some workaround for this, like downgrading to the previous agent.jar perhaps?</p>

<p>I've taken a number of thread dumps of hanging agent.jar processes. They're all slightly different. What they all have in common is multiple threads stuck waiting:<br/>
 at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)</p>

<p>I'll upload a couple of the dumps, but they're really all much the same as what <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kapawo" class="user-hover" rel="kapawo">kapawo</a> has uploaded.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER141899's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424078&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424078">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>We are also hitting this while testing upgrade to 2.332.1 from 2.319.3 . We are hitting this issue when trying to add agents via the swarm-plugin. Our pipeline hangs in the node step and this deadlock is found on the agent's side:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Found one Java-level deadlock:
=============================
"pool-1-thread-16":
  waiting to lock monitor 0x00007f2f04007ab8 (object 0x000000008d73ebe8, a hudson.remoting.RemoteClassLoader),
  which is held by "pool-1-thread-8 / waiting for JNLP4-connect connection to &lt;Jenkins Controller&gt; id=58"
"pool-1-thread-8 / waiting for JNLP4-connect connection to &lt;Jenkins Controller&gt; id=58":
  waiting to lock monitor 0x00007f2f040035f8 (object 0x00000000f1d1ba30, a hudson.util.RingBufferLogHandler),
  which is held by "pool-1-thread-16"


Java stack information for the threads listed above:
===================================================
"pool-1-thread-16":
	at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
	- locked &lt;0x00000000f1d1ba30&gt; (a hudson.util.RingBufferLogHandler)
	at java.util.logging.Logger.log(Logger.java:738)
	at java.util.logging.Logger.doLog(Logger.java:765)
	at java.util.logging.Logger.log(Logger.java:831)
	at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Writer.run(BIONetworkLayer.java:184)
	- locked &lt;0x000000008d6cfea8&gt; (a org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Writer)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:122)
	at hudson.remoting.Engine$1$$Lambda$12/910059620.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:748)
"pool-1-thread-8 / waiting for JNLP4-connect connection to &lt;Jenkins Controller&gt; id=58":
	at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)	
- waiting to lock &lt;0x00000000f1d1ba30&gt; (a hudson.util.RingBufferLogHandler)
	at java.util.logging.Logger.log(Logger.java:738)
	at java.util.logging.Logger.doLog(Logger.java:765)
	at java.util.logging.Logger.log(Logger.java:851)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processWrite(SSLEngineFilterLayer.java:486)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processQueuedWrites(SSLEngineFilterLayer.java:246)
	- locked &lt;0x000000008d6cfc08&gt; (a java.lang.Object)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doSend(SSLEngineFilterLayer.java:198)
	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.doSend(ProtocolStack.java:700)
	at org.jenkinsci.remoting.protocol.ApplicationLayer.write(ApplicationLayer.java:156)
	at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer$ByteBufferCommandTransport.write(ChannelApplicationLayer.java:325)
	at hudson.remoting.AbstractByteBufferCommandTransport.write(AbstractByteBufferCommandTransport.java:301)
	at hudson.remoting.Channel.send(Channel.java:766)
	- locked &lt;0x000000008d6a2ef8&gt; (a hudson.remoting.Channel)
	at hudson.remoting.Request.call(Request.java:167)
	- locked &lt;0x00000000dd8c3f08&gt; (a hudson.remoting.RemoteInvocationHandler$RPCRequest)
	- locked &lt;0x000000008d6a2ef8&gt; (a hudson.remoting.Channel)
	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:289)
	at com.sun.proxy.$Proxy6.fetch3(Unknown Source)
	at hudson.remoting.RemoteClassLoader.prefetchClassReference(RemoteClassLoader.java:348)
	at hudson.remoting.RemoteClassLoader.loadWithMultiClassLoader(RemoteClassLoader.java:253)
	at hudson.remoting.RemoteClassLoader.findClass(RemoteClassLoader.java:223)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	- locked &lt;0x000000008d73ebe8&gt; (a hudson.remoting.RemoteClassLoader)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.getDeclaringClass0(Native Method)
	at java.lang.Class.getDeclaringClass(Class.java:1235)
	at java.lang.Class.getEnclosingClass(Class.java:1277)
	at java.lang.Class.getSimpleBinaryName(Class.java:1443)
	at java.lang.Class.getSimpleName(Class.java:1309)
	at java.lang.Class.isAnonymousClass(Class.java:1411)
	at org.jenkinsci.remoting.util.AnonymousClassWarnings.doCheck(AnonymousClassWarnings.java:76)
	at org.jenkinsci.remoting.util.AnonymousClassWarnings.lambda$check$0(AnonymousClassWarnings.java:66)
	at org.jenkinsci.remoting.util.AnonymousClassWarnings$$Lambda$26/541145666.run(Unknown Source)
	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)
	at hudson.remoting.InterceptingExecutorService$$Lambda$27/1987554325.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:122)
	at hudson.remoting.Engine$1$$Lambda$12/910059620.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:748)


Found 1 deadlock. </pre>
</div></div>
<p>Looking at the code <a href="https://github.com/jenkinsci/remoting/blob/master/src/main/java/org/jenkinsci/remoting/protocol/impl/SSLEngineFilterLayer.java#L486" class="external-link" target="_blank" rel="nofollow noopener">SSLEngineFilterLayer#L486</a> and <a href="https://github.com/jenkinsci/remoting/blob/master/src/main/java/org/jenkinsci/remoting/protocol/impl/BIONetworkLayer.java#L184" class="external-link" target="_blank" rel="nofollow noopener">BIONetworkLayer.java#L184</a> seems to me that this problem presents itself when logLevel is set to <b>FINEST</b>. I have been able to work around it by modifying my logging.properties.</p>
</pre>
</details>

</details>
<p>We are also hitting this while testing upgrade to 2.332.1 from 2.319.3 . We are hitting this issue when trying to add agents via the swarm-plugin. Our pipeline hangs in the node step and this deadlock is found on the agent's side:</p>


```
Found one Java-level deadlock:
=============================
"pool-1-thread-16":
  waiting to lock monitor 0x00007f2f04007ab8 (object 0x000000008d73ebe8, a hudson.remoting.RemoteClassLoader),
  which is held by "pool-1-thread-8 / waiting for JNLP4-connect connection to <Jenkins Controller> id=58"
"pool-1-thread-8 / waiting for JNLP4-connect connection to <Jenkins Controller> id=58":
  waiting to lock monitor 0x00007f2f040035f8 (object 0x00000000f1d1ba30, a hudson.util.RingBufferLogHandler),
  which is held by "pool-1-thread-16"


Java stack information for the threads listed above:
===================================================
"pool-1-thread-16":
	at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)
	- locked <0x00000000f1d1ba30> (a hudson.util.RingBufferLogHandler)
	at java.util.logging.Logger.log(Logger.java:738)
	at java.util.logging.Logger.doLog(Logger.java:765)
	at java.util.logging.Logger.log(Logger.java:831)
	at org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Writer.run(BIONetworkLayer.java:184)
	- locked <0x000000008d6cfea8> (a org.jenkinsci.remoting.protocol.impl.BIONetworkLayer$Writer)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:122)
	at hudson.remoting.Engine$1$$Lambda$12/910059620.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:748)
"pool-1-thread-8 / waiting for JNLP4-connect connection to <Jenkins Controller> id=58":
	at hudson.util.RingBufferLogHandler.publish(RingBufferLogHandler.java:78)	
- waiting to lock <0x00000000f1d1ba30> (a hudson.util.RingBufferLogHandler)
	at java.util.logging.Logger.log(Logger.java:738)
	at java.util.logging.Logger.doLog(Logger.java:765)
	at java.util.logging.Logger.log(Logger.java:851)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processWrite(SSLEngineFilterLayer.java:486)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processQueuedWrites(SSLEngineFilterLayer.java:246)
	- locked <0x000000008d6cfc08> (a java.lang.Object)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doSend(SSLEngineFilterLayer.java:198)
	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.doSend(ProtocolStack.java:700)
	at org.jenkinsci.remoting.protocol.ApplicationLayer.write(ApplicationLayer.java:156)
	at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer$ByteBufferCommandTransport.write(ChannelApplicationLayer.java:325)
	at hudson.remoting.AbstractByteBufferCommandTransport.write(AbstractByteBufferCommandTransport.java:301)
	at hudson.remoting.Channel.send(Channel.java:766)
	- locked <0x000000008d6a2ef8> (a hudson.remoting.Channel)
	at hudson.remoting.Request.call(Request.java:167)
	- locked <0x00000000dd8c3f08> (a hudson.remoting.RemoteInvocationHandler$RPCRequest)
	- locked <0x000000008d6a2ef8> (a hudson.remoting.Channel)
	at hudson.remoting.RemoteInvocationHandler.invoke(RemoteInvocationHandler.java:289)
	at com.sun.proxy.$Proxy6.fetch3(Unknown Source)
	at hudson.remoting.RemoteClassLoader.prefetchClassReference(RemoteClassLoader.java:348)
	at hudson.remoting.RemoteClassLoader.loadWithMultiClassLoader(RemoteClassLoader.java:253)
	at hudson.remoting.RemoteClassLoader.findClass(RemoteClassLoader.java:223)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	- locked <0x000000008d73ebe8> (a hudson.remoting.RemoteClassLoader)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at java.lang.Class.getDeclaringClass0(Native Method)
	at java.lang.Class.getDeclaringClass(Class.java:1235)
	at java.lang.Class.getEnclosingClass(Class.java:1277)
	at java.lang.Class.getSimpleBinaryName(Class.java:1443)
	at java.lang.Class.getSimpleName(Class.java:1309)
	at java.lang.Class.isAnonymousClass(Class.java:1411)
	at org.jenkinsci.remoting.util.AnonymousClassWarnings.doCheck(AnonymousClassWarnings.java:76)
	at org.jenkinsci.remoting.util.AnonymousClassWarnings.lambda$check$0(AnonymousClassWarnings.java:66)
	at org.jenkinsci.remoting.util.AnonymousClassWarnings$$Lambda$26/541145666.run(Unknown Source)
	at hudson.remoting.InterceptingExecutorService.lambda$wrap$0(InterceptingExecutorService.java:78)
	at hudson.remoting.InterceptingExecutorService$$Lambda$27/1987554325.call(Unknown Source)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:122)
	at hudson.remoting.Engine$1$$Lambda$12/910059620.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:748)


Found 1 deadlock. ```
<p>Looking at the code <a href="https://github.com/jenkinsci/remoting/blob/master/src/main/java/org/jenkinsci/remoting/protocol/impl/SSLEngineFilterLayer.java#L486" class="external-link" target="_blank" rel="nofollow noopener">SSLEngineFilterLayer#L486</a> and <a href="https://github.com/jenkinsci/remoting/blob/master/src/main/java/org/jenkinsci/remoting/protocol/impl/BIONetworkLayer.java#L184" class="external-link" target="_blank" rel="nofollow noopener">BIONetworkLayer.java#L184</a> seems to me that this problem presents itself when logLevel is set to <b>FINEST</b>. I have been able to work around it by modifying my logging.properties.</p>

<!-- ### Imported Jira references for easier searching -->
<!-- [jira_issue_key=JENKINS-68122] -->
<!-- [jira_comment_id=424078] -->
<!-- [comment_author=JIRAUSER141899] -->

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER142437's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424083&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424083">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jorgetm" class="user-hover" rel="jorgetm">jorgetm</a> </p>

<p>Our usage that causes the hang is also adding of agents via swarm plugin.</p>

<p>About your workaround, we haven't done anything with logging.properties, so we should be running at the JDK default INFO level, and we're still getting the deadlocks.</p>

<p>I'm curious exactly what you did to work around this bug.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jorgetm" class="user-hover" rel="jorgetm">jorgetm</a> </p>

<p>Our usage that causes the hang is also adding of agents via swarm plugin.</p>

<p>About your workaround, we haven't done anything with logging.properties, so we should be running at the JDK default INFO level, and we're still getting the deadlocks.</p>

<p>I'm curious exactly what you did to work around this bug.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER141899's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424084&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424084">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=roband7" class="user-hover" rel="roband7">roband7</a> In our case, we pass in `-Djava.util.logging.config.file` when we launch the swarm-client.jar. The logging.properties file we pass in has historically set the log level to <b>ALL</b> since we have hit issues in the past with the client and wanted to have logs for debugging our swarm agents. I modified the level to <b>CONFIG</b> and haven't hit the issue again (have spun up a few affected builds successfully, will keep trying to reproduce, but before the agents would hang on every build). At first, I tried <b>FINE</b> but experienced some hangs with that so downgraded even further.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=roband7" class="user-hover" rel="roband7">roband7</a> In our case, we pass in `-Djava.util.logging.config.file` when we launch the swarm-client.jar. The logging.properties file we pass in has historically set the log level to <b>ALL</b> since we have hit issues in the past with the client and wanted to have logs for debugging our swarm agents. I modified the level to <b>CONFIG</b> and haven't hit the issue again (have spun up a few affected builds successfully, will keep trying to reproduce, but before the agents would hang on every build). At first, I tried <b>FINE</b> but experienced some hangs with that so downgraded even further.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER142437's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424085&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424085">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>My guess is that something is just iffy with logging and thread synchronization in the agent, and that different log levels and different usage patterns just happen to have different risks of hitting the bug.</p>

<p>I'll try changing the log level from INFO to WARNING just to lower the risk even more.</p>
</pre>
</details>

</details>
<p>My guess is that something is just iffy with logging and thread synchronization in the agent, and that different log levels and different usage patterns just happen to have different risks of hitting the bug.</p>

<p>I'll try changing the log level from INFO to WARNING just to lower the risk even more.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jglick's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424141&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424141">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Whereas I thought I knew what was going on in the original thread dumps, that hypothesis is apparently violated by the thread dump excerpts you show (is there anything else at all related in the full thread dumps? <tt>Found one Java-level deadlock</tt>?) and it does not seem to make any sense for the new line 78</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
LogRecordRef logRecordRef = <span class="code-keyword">new</span> LogRecordRef(record);
</pre>
</div></div>

<p>to be blocked on any object monitor. Why introducing a separate lock object would help in this context, I cannot guess, so it is unclear that moving the <tt>LogRecordRef</tt> constructor (which I hypothesized triggered class loading) out of the <tt>synchronized</tt> block is even necessary. IOW this might work just as well:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">
    <span class="code-keyword">public</span> void publish(LogRecord record) {
        <span class="code-keyword">synchronized</span> (lock) {
            <span class="code-object">int</span> len = records.length;
            records[(start + size) % len] = <span class="code-keyword">new</span> LogRecordRef(record);
            <span class="code-keyword">if</span> (size == len) {
                start = (start + 1) % len;
            } <span class="code-keyword">else</span> {
                size++;
            }
        }
    }
</pre>
</div></div>
</pre>
</details>

</details>
<p>Whereas I thought I knew what was going on in the original thread dumps, that hypothesis is apparently violated by the thread dump excerpts you show (is there anything else at all related in the full thread dumps? <tt>Found one Java-level deadlock</tt>?) and it does not seem to make any sense for the new line 78</p>


<pre>

LogRecordRef logRecordRef = <span class="code-keyword">new</span> LogRecordRef(record);
</pre>

<p>to be blocked on any object monitor. Why introducing a separate lock object would help in this context, I cannot guess, so it is unclear that moving the <tt>LogRecordRef</tt> constructor (which I hypothesized triggered class loading) out of the <tt>synchronized</tt> block is even necessary. IOW this might work just as well:</p>


<pre>

    <span class="code-keyword">public</span> void publish(LogRecord record) {
<span class="code-keyword">synchronized</span> (lock) {
    <span class="code-object">int</span> len = records.length;
    records[(start + size) % len] = <span class="code-keyword">new</span> LogRecordRef(record);
    <span class="code-keyword">if</span> (size == len) {
start = (start + 1) % len;
    } <span class="code-keyword">else</span> {
size++;
    }
}
    }
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER142437's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424165&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424165">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I have done some testing with all of your patches. I'm still getting deadlocks.</p>

<p>Looking at this some more, and experimenting with my own patches.</p>

<p>My conclusion is that it doesn't matter at all how much we change the synchronization here.</p>

<p>The only thing that matters is the fact that instantiation of a LogRecordRef inside the publish method triggers the classloader.</p>

<p>So any thread that has the classloader locked, and does something that waits for another thread, the other thread does some logging, and hits publish that instantiates LogRecordRef that tries to lock the classloader, boom deadlock.</p>

<p>Uploading a threaddump showing this, this was taken with the combined patches from Jesse and Basil.</p>

<p>One hackish fix for this issue is to just instantiate a dummy LogRecordRef in the RingBufferLogHandler constructor. I'm sure there's better ways.</p>
</pre>
</details>

</details>
<p>I have done some testing with all of your patches. I'm still getting deadlocks.</p>

<p>Looking at this some more, and experimenting with my own patches.</p>

<p>My conclusion is that it doesn't matter at all how much we change the synchronization here.</p>

<p>The only thing that matters is the fact that instantiation of a LogRecordRef inside the publish method triggers the classloader.</p>

<p>So any thread that has the classloader locked, and does something that waits for another thread, the other thread does some logging, and hits publish that instantiates LogRecordRef that tries to lock the classloader, boom deadlock.</p>

<p>Uploading a threaddump showing this, this was taken with the combined patches from Jesse and Basil.</p>

<p>One hackish fix for this issue is to just instantiate a dummy LogRecordRef in the RingBufferLogHandler constructor. I'm sure there's better ways.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER142437's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424171&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424171">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>That constructor hack is in fact the exact same workaround I ended up with <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>We've been running it for a few hours now on our production Jenkins server, so far not a single hang. Looking good.</p>
</pre>
</details>

</details>
<p>That constructor hack is in fact the exact same workaround I ended up with <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p>We've been running it for a few hours now on our production Jenkins server, so far not a single hang. Looking good.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>basil's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424172&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424172">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I agree that we should preload <tt>LogRecordRef</tt> on the agent side to avoid an ABBA deadlock. But rather than doing this in the constructor inside the agent JVM, I wonder if it might be cleaner to do this inside a static initializer as in <a href="https://github.com/basil/jenkins/commit/38efb0b99c20376c2dee27050b00530bda3fe776" class="external-link" target="_blank" rel="nofollow noopener">commit 38efb0b99c</a>. That way it only needs to be done once when <tt>RingBufferLogHandler</tt> is loaded on the agent rather than every time a new instance of <tt>RingBufferLogHandler</tt> is created. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=roband7" class="user-hover" rel="roband7">roband7</a> Would you be interested in testing this commit?</p>
</pre>
</details>

</details>
<p>I agree that we should preload <tt>LogRecordRef</tt> on the agent side to avoid an ABBA deadlock. But rather than doing this in the constructor inside the agent JVM, I wonder if it might be cleaner to do this inside a static initializer as in <a href="https://github.com/basil/jenkins/commit/38efb0b99c20376c2dee27050b00530bda3fe776" class="external-link" target="_blank" rel="nofollow noopener">commit 38efb0b99c</a>. That way it only needs to be done once when <tt>RingBufferLogHandler</tt> is loaded on the agent rather than every time a new instance of <tt>RingBufferLogHandler</tt> is created. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=roband7" class="user-hover" rel="roband7">roband7</a> Would you be interested in testing this commit?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER142437's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424174&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424174">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Sure, if Jesse agrees with that fix I'll compile and install it on our production server later today. Cannot do it right away, been enough disruptions since the 2.332.1 upgrade.</p>

<p>Our setup hits the bug fast due to heavy agent usage, so will know soon if the new fix works too.</p>
</pre>
</details>

</details>
<p>Sure, if Jesse agrees with that fix I'll compile and install it on our production server later today. Cannot do it right away, been enough disruptions since the 2.332.1 upgrade.</p>

<p>Our setup hits the bug fast due to heavy agent usage, so will know soon if the new fix works too.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jglick's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424176&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424176">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Well I hope the fix works this time and it can go into 2.332.3. Since this is pretty severe for those who encounter it, I took a stab at an agent-side workaround. Completely untested, just based on code inspection, but if it works then it might allow you to use a stock core for the next few weeks: <a href="https://github.com/jenkinsci/remoting/pull/527" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/remoting/pull/527</a></p>
</pre>
</details>

</details>
<p>Well I hope the fix works this time and it can go into 2.332.3. Since this is pretty severe for those who encounter it, I took a stab at an agent-side workaround. Completely untested, just based on code inspection, but if it works then it might allow you to use a stock core for the next few weeks: <a href="https://github.com/jenkinsci/remoting/pull/527" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/remoting/pull/527</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>basil's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=424288&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-424288">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Fixed in <a href="https://github.com/jenkinsci/jenkins/pull/6449" class="external-link" target="_blank" rel="nofollow noopener">jenkinsci/jenkins#6449</a> toward 2.343.</p>
</pre>
</details>

</details>
<p>Fixed in <a href="https://github.com/jenkinsci/jenkins/pull/6449" class="external-link" target="_blank" rel="nofollow noopener">jenkinsci/jenkins#6449</a> toward 2.343.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER139016's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=426004&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-426004">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I read through the comments here and didn't see any workaround for until the new LTS is released.  We are running on 2.332.3 and have this problem BAD.  We are failing jobs several times per day.  Is there any workaround that does not involve leaving the LTS line?</p>
</pre>
</details>

</details>
<p>I read through the comments here and didn't see any workaround for until the new LTS is released.  We are running on 2.332.3 and have this problem BAD.  We are failing jobs several times per day.  Is there any workaround that does not involve leaving the LTS line?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>markewaite's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=426007&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-426007">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=koyaanisqatsi" class="user-hover" rel="koyaanisqatsi">koyaanisqatsi</a> the earlier comments indicate that reducing the logging level of the agent seems to reduce the issue.  Have you tried that?</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=koyaanisqatsi" class="user-hover" rel="koyaanisqatsi">koyaanisqatsi</a> the earlier comments indicate that reducing the logging level of the agent seems to reduce the issue.  Have you tried that?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER139016's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=426069&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-426069">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> I'm not sure how to do that on our installation.  We are not launching agents from a CLI.  They are launched automatically in AWS by "Clouds and Nodes" configs.  I'm trying to identify the plugin that provides this, which I believe is this guy: <a href="https://plugins.jenkins.io/ssh-slaves/" class="external-link" target="_blank" rel="nofollow noopener">https://plugins.jenkins.io/ssh-slaves/</a>  I have not found any way to inject a logging config into our agent startup process.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> I'm not sure how to do that on our installation.  We are not launching agents from a CLI.  They are launched automatically in AWS by "Clouds and Nodes" configs.  I'm trying to identify the plugin that provides this, which I believe is this guy: <a href="https://plugins.jenkins.io/ssh-slaves/" class="external-link" target="_blank" rel="nofollow noopener">https://plugins.jenkins.io/ssh-slaves/</a>  I have not found any way to inject a logging config into our agent startup process.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>markewaite's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=426070&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-426070">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>If you're not increasing the logging level, then there is likely no benefit to reducing the logging level.</p>

<p>You could assist with testing the <a href="https://groups.google.com/g/jenkinsci-dev/c/TQsZFHYLWhk/m/mpg4KnD5BgAJ" class="external-link" target="_blank" rel="nofollow noopener">Jenkins 2.346.1 release candidate </a> and confirm that it is resolved in your environment with that release.</p>
</pre>
</details>

</details>
<p>If you're not increasing the logging level, then there is likely no benefit to reducing the logging level.</p>

<p>You could assist with testing the <a href="https://groups.google.com/g/jenkinsci-dev/c/TQsZFHYLWhk/m/mpg4KnD5BgAJ" class="external-link" target="_blank" rel="nofollow noopener">Jenkins 2.346.1 release candidate </a> and confirm that it is resolved in your environment with that release.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>basil's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=426071&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-426071">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>, you appear to be confused about which LTS release this fix was backported to. The fix was backported to 2.332.3, so testing 2.346.1 will not help. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=koyaanisqatsi" class="user-hover" rel="koyaanisqatsi">koyaanisqatsi</a> I suggest filing a new ticket, as your problem may be unrelated.</p>
</pre>
</details>

</details>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>, you appear to be confused about which LTS release this fix was backported to. The fix was backported to 2.332.3, so testing 2.346.1 will not help. <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=koyaanisqatsi" class="user-hover" rel="koyaanisqatsi">koyaanisqatsi</a> I suggest filing a new ticket, as your problem may be unrelated.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>JIRAUSER141913's <a href="https://issues.jenkins.io/browse/JENKINS-68122?focusedId=426086&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-426086">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>we no longer has this issue after upgrade to 2.332.3. Matt Chester, your issue may be different. However, you can try our work around (earlier comment) which we used before the 2.332.3 LTS released. </p>
</pre>
</details>

</details>
<p>we no longer has this issue after upgrade to 2.332.3. Matt Chester, your issue may be different. However, you can try our work around (earlier comment) which we used before the 2.332.3 LTS released. </p>
